drivers/gpio/gpio_nrfe.c

```diff
@@ -0,0 +1,158 @@
+		return ret_val;
+	}
+
+	msg.opcode = NRFE_GPIO_PIN_CLEAR;
+	msg.pin = clear_mask;
+
+	return ipc_service_send(&ep, (void *)(&msg), sizeof(nrfe_gpio_data_packet_t));
+}
+
+static int gpio_nrfe_port_set_bits_raw(const struct device *port, gpio_port_pins_t mask)
+{
+	nrfe_gpio_data_packet_t msg = {
+		.opcode = NRFE_GPIO_PIN_SET, .pin = mask, .port = get_port_cfg(port)->port_num};
+
+	return ipc_service_send(&ep, (void *)(&msg), sizeof(nrfe_gpio_data_packet_t));
+}
+
+static int gpio_ nrfe_port_clear_bits_raw(const struct device *port, gpio_port_pins_t mask)
+{
+	nrfe_gpio_data_packet_t msg = {
+		.opcode = NRFE_GPIO_PIN_CLEAR, .pin = mask, .port = get_port_cfg(port)->port_num};
+
+	return ipc_service_send(&ep, (void *)(&msg), sizeof(nrfe_gpio_data_packet_t));
+}
+
+static int gpio_nrfe_port_toggle_bits(const struct device *port, gpio_port_pins_t mask)
+{
+	nrfe_gpio_data_packet_t msg = {
+		.opcode = NRFE_GPIO_PIN_TOGGLE, .pin = mask, .port = get_port_cfg(port)->port_num};
+
+	return ipc_service_send(&ep, (void *)(&msg), sizeof(nrfe_gpio_data_packet_t));
+}
+
+static int gpio_nrfe_init(const struct device *port)
+{
+	const struct device *ipc0_instance = DEVICE_DT_GET(DT_NODELABEL(ipc0));
+	int ret = ipc_service_open_instance(ipc0_instance);
+
+	if ((ret < 0) && (ret != -EALREADY)) {
+		return ret;
+	}
+
+	ret = ipc_service_register_endpoint(ipc0_instance, &ep, &ep_cfg);
+	if (ret < 0) {
+		return ret;
+	}
+
+	k_sem_take(&bound_sem, K_FOREVER);
+
+	return 0;
+}
+
+static const struct gpio_driver_api gpio_nrfe_drv_api_funcs = {
+	.pin_configure = gpio_nrfe_pin_configure,
+	.port_set_masked_raw = gpio_nrfe_port_set_masked_raw,
+	.port_set_bits_raw = gpio_nrfe_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_nrfe_port_clear_bits_raw,
+	.port_toggle_bits = gpio_nrfe_port_toggle_bits,
+};
+
+#define GPIO_NRFE_DEVICE(id)                                                                       \
+	static const struct gpio_nrfe_cfg gpio_nrfe_p##id##_cfg = {                                \
+		.common =                                                                          \
+			{                                                                          \
+				.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(id),              \
+			},                                                                         \
+		.port_num = DT_INST_PROP(id, port),                                                \
+	};                                                                                         \
+                                                                                                   \
+	static struct gpio_nrfe_data gpio_nrfe_p##id##_data;                                       \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(id, gpio_nrfe_init, NULL, &gpio_nrfe_p##id##_data,                   \
+			      &gpio_nrfe_p##id##_cfg, POST_KERNEL, CONFIG_GPIO_INIT_PRIORITY,      \
```

Comment:
Is is always true that GPIO init priority is lower than ipc_service init priority?
Otherwise, wouldn't that cause some issues?

Reply:
I added another commit that changes the default init priority when eGPIO is enabled, so that it was initialized after IPC.